src/dswx_sar/stage_pekel_water.py

```diff
@@ -0,0 +1,539 @@
+    length = ds.GetRasterBand(1).YSize
+    width = ds.GetRasterBand(1).XSize
+    input_y_min = input_y_max + (length * yres)
+    input_x_max = input_x_min + (width * xres)
+
+    x_min = max(x_min, input_x_min)
+    x_max = min(x_max, input_x_max)
+    y_min = max(y_min, input_y_min)
+    y_max = min(y_max, input_y_max)
+
+    gdal.Translate(
+        output_path, ds, format='GTiff', projWin=[x_min, y_max, x_max, y_min]
+    )
+
+
+def download_reference_water_map(polys,
+                                 watermap_source,
+                                 watermap_bucket,
+                                 watermap_type,
+                                 outfile):
+    """
+    Download a Pekel's map from the s3 bucket.
+
+    Parameters:
+    ----------
+    polys: list of shapely.geometry.Polygon
+        List of shapely polygons.
+    watermap_source : str
+        Name of the S3 bucket containing the full Worldcover map to download from.
+    watermap_type : str
+        Type of Pekel's water map (i.e., seasonality, occurrence)
+    outfile:
+        Path to the where the output reference water file is to be staged.
+    """
+    # Download Pekel's water map for each polygon/epsg
+    file_prefix = os.path.splitext(outfile)[0]
+    wc_list = []
+    counter = 0
+    for idx, poly in enumerate(polys):
+        x_min, y_min, x_max, y_max = poly.bounds
+
+        if watermap_source == 'googleapi':
+            x_min_10 = int((x_min // 10) * 10)
+            y_min_10 = int((y_min // 10) * 10)
+            x_max_10 = int(((x_max // 10) + 1) * 10)
+            y_max_10 = int(((y_max // 10) + 1) * 10)
+            # Pekel's water map is only available from 50S to 90N.
+            x_min_10 = np.max([x_min_10, -180])
+            x_max_10 = np.min([x_max_10, 180])
+            y_min_10 = np.max([y_min_10, -50])
+            y_max_10 = np.min([y_max_10, 90])
+
+            # Pekel's water map is stored as 10 x 10 deg.
+            for x_10 in range(x_min_10, x_max_10, 10):
+                for y_10 in range(y_min_10, y_max_10, 10):
+                    w_or_e = 'W' if x_10 < 0 else 'E'
+                    s_or_n = 'S' if y_10 < 0 else 'N'
+
+                    x_min_10_str = int(np.abs(x_10))
+                    y_min_10_str = int(np.abs(y_10))
+                    google_api_filename = f'{watermap_type}_{x_min_10_str}{w_or_e}_{y_min_10_str}{s_or_n}v1_4_2021.tif'
+                    
+                    url = os.path.join(PEKEL_GOOGLE_SOURCE, watermap_type, google_api_filename)
+                    code = urllib.request.urlopen(url).getcode()
+                    if (code != 404):
+                        logger.info(f"Downloading {url} ({str(counter)})")
+                        urllib.request.urlretrieve(url, google_api_filename)
+                        wc_list.append(google_api_filename)
+                    else:
+                        logger.info(f"{url} not found")
+                    counter += 1
+        else:
```

Comment:
Based on what I see in here and `check_aws_connection()`,It looks like the code would only work with the google public storage. No need to add the functionality to make it work with S3 bucket, but wanted to check with you.